dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsConnector.java

```diff
@@ -393,6 +398,39 @@ public TeradataAssessmentLogsJdbcTask(@Nonnull String targetPath, SharedState st
             super(targetPath, state, logTable, queryTable, conditions, interval, orderBy);
         }
 
+        //Need to be overridden because for byte[] array toString returns not string representation but address of array in memory
+        @Nonnull
+        @Override
+        protected ResultSetExtractor<Void> newCsvResultSetExtractor(@Nonnull ByteSink sink, long count) {
+            return rs -> {
+                CSVFormat format = newCsvFormat(rs);
+                try (RecordProgressMonitor monitor
+                             = count >= 0
+                        ? new RecordProgressMonitor(getName(), count)
+                        : new RecordProgressMonitor(getName());
+                     Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream();
+                     CSVPrinter printer = format.print(writer)) {
+                    final int columnCount = rs.getMetaData().getColumnCount();
+                    while (rs.next()) {
+                        monitor.count();
+                        for (int i = 1; i <= columnCount; i++) {
+                            Object object = rs.getObject(i);
+                            if (object instanceof byte[]) {
+                                printer.print(Base64.getEncoder().encodeToString((byte[]) object));
```

Comment:
I think I'd like to see this, and potentially the Clob path from Oracle, both pushed into the superclass. I think this is a reasonable good behaviour for the common code. What do other reviewers think?

Reply:
I'd suggest to do it in next PR

Reply:
If we know that a 'next' PR is going to happen on some sort of deterministic timeline, that's OK, but if this pushes the refactor off into the indefinite future, then I'd push us to do this now before the methods have a chance to diverge further.

Reply:
ok, it is done in this PR, please see a9a7a4be060a487ee2d299a0478dc569e18c1436